File: 프로그래머스/평행.py
dots =[[4, 1], [4, 1], [2, 4], [5, 10]]	
slopes = set()
slope_list = []
for i in range(3):
    x1 = dots[i][0]
    y1 = dots[i][1]
    for j in range(i+1,4):
        print(f"i = {i}")
        print(f"j = {j}")
        x2 = dots[j][0]
        y2 = dots[j][1]
        if x1==x2 and y1==y2:
            slope = f"{i}{j}"
        else:
            slope = (y1-y2)/(x1-x2)
        slopes.add(slope)
        slope_list.append(slope)

answer = 1 if len(slopes) == 6 else 0     
print(answer)

File: 프로그래머스/평행.js
function solution(dots) {
    let slopes = new Set();
    // 두 선분이 평행하다 = 두 선분의 기울기가 같다
    // 선분의 기울기 = (y1-y2) / (x1-x2)
    for (let i = 0; i < 4; i++) {
        let x1 = dots[i][0];
        let y1 = dots[i][1];
        for (let j = i + 1; j < 4; j++) {
            console.log(`i = ${i}, j = ${j}`)
            let x2 = dots[j][0];
            let y2 = dots[j][1];
            if (x1 === x2 && y1 === y2) {
                slope = `${i}${j}`;
            }
            else {
                let slope = (y1 - y2) / (x1 - x2);
            }
            slopes.add(slope);

        }
    }
    answer = slopes.size === 9 ? 0 : 1;
    return answer;
}

let dots = [[4, 1], [4, 1], [2, 4], [5, 10]];

console.log(solution(dots));